certifikit-cli/build.gradle

```diff
@@ -6,59 +6,44 @@ plugins {
   id 'com.palantir.graal' version '0.7.1'
 }
 
-shadowJar {
+application {
   mainClassName = 'app.cash.certifikit.cli.Main'
 }
 
+shadowJar {
+  mainClassName = application.mainClassName
+}
+
 jar {
   manifest {
     attributes 'Automatic-Module-Name': 'app.cash.certifikit.cli'
-    attributes 'Main-Class': 'app.cash.certifikit.cli.Main'
+    attributes 'Main-Class': application.mainClassName
   }
 }
 
+
 //https://github.com/remkop/picocli/tree/master/picocli-codegen
 configurations {
   generateConfig
 }
 
-dependencies {
-  api project(':certifikit')
-  api dep.okio
-  implementation 'info.picocli:picocli:4.4.0'
-  implementation 'com.squareup.okhttp3:okhttp:4.8.0'
-  implementation 'com.squareup.okhttp3:okhttp-tls:4.8.0'
-
-  annotationProcessor 'info.picocli:picocli-codegen:4.4.0'
-  generateConfig 'info.picocli:picocli-codegen:4.4.0'
-
-  compileOnly("org.graalvm.nativeimage:svm:20.1.0")
-
-  testImplementation dep.junit
-  testImplementation dep.assertj
-}
-
-application {
-  mainClassName('app.cash.certifikit.cli.Main')
-}
+task(generateGraalReflectionConfig, dependsOn: 'classes', group:'graal', type: JavaExec) {
+  def reflectionOutputFile = "${buildDir}/resources/main/META-INF/native-image/${project.group}/${project.name}/reflect-config.json"
 
-def reflectionOutputFile = "${buildDir}/resources/main/META-INF/native-image/${project.group}/${project.name}/reflect-config.json"
+  description 'Generates a temporary graal reflecftion config for native image.'
+  outputs.file(reflectionOutputFile)
 
-task(generateGraalReflectionConfig, dependsOn: 'classes', type: JavaExec) {
   main = 'picocli.codegen.aot.graalvm.ReflectionConfigGenerator'
   classpath = configurations.generateConfig + sourceSets.main.runtimeClasspath
-  args = ["--output=$reflectionOutputFile", 'app.cash.certifikit.cli.Main']
+  args = ["--output=$reflectionOutputFile", application.mainClassName]
 }
-assemble.dependsOn generateGraalReflectionConfig
-assemble.dependsOn installShadowDist
 
 graal {
-  mainClass("app.cash.certifikit.cli.Main")
+  mainClass(application.mainClassName)
   outputName("cft")
   graalVersion('20.1.0')
   javaVersion('11')
 
-  option("-H:ReflectionConfigurationFiles=${reflectionOutputFile}")
```

Comment:
This is cleanup, pulled in via being included in JAR for a specific known path already